src/fsharp/fsc.fs

```diff
@@ -1840,30 +1861,38 @@ let main1(tcGlobals, tcImports: TcImports, frameworkTcImports, generatedCcu, typ
       use unwindBuildPhase = PushThreadBuildPhaseUntilUnwind (BuildPhase.Output)    
       if tcConfig.printSignature   then InterfaceFileWriter.WriteInterfaceFile (tcGlobals,tcConfig, InfoReader(tcGlobals,tcImports.GetImportMap()), typedAssembly);
       ReportTime tcConfig ("Write XML document signatures")
-      if tcConfig.xmlDocOutputFile.IsSome then 
+      match xmlDocOutputOpt with
+      | Some(_) ->
```

Comment:
converted from `if` to `match` because i need type inference, otherwise i must specify method signature for `xmlDocOutputOpt` param (more invasive diff)